File: Classical/src/main/java/prv/zielony/transaction/analyst/classical/model/ContactData.java
package prv.zielony.transaction.analyst.classical.model;

import javax.persistence.Embeddable;
import java.util.LinkedList;
import java.util.List;

/**
 * Created by zielony on 26.02.16.
 */
@Embeddable
public class ContactData {

  private List<String> phoneNumbers = new LinkedList<String>();

  private List<String> emails = new LinkedList<String>();

  private List<Address> addresses = new LinkedList<Address>();

    public List<String> getPhoneNumbers() {
        return phoneNumbers;
    }

    public void setPhoneNumbers(List<String> phoneNumbers) {
        this.phoneNumbers = phoneNumbers;
    }

    public List<String> getEmails() {
        return emails;
    }

    public void setEmails(List<String> emails) {
        this.emails = emails;
    }

    public List<Address> getAddresses() {
        return addresses;
    }

    public void setAddresses(List<Address> addresses) {
        this.addresses = addresses;
    }
}

File: Modern/src/main/scala/prv/zielony/transaction/analyst/modern/events/users/ContactData.scala
package prv.zielony.transaction.analyst.modern.events.users

/**
 * Created by zielony on 20.02.16.
 */
case class ContactData(val phoneNumbers: Traversable[String] = Traversable(),
                       val emails:Traversable[String] = Traversable(),
                       val addresses:Traversable[Address] = Traversable())

case class Address(val street:Option[Street], val city:Option[String], val country:Option[String]);

case class Street(val name:String, val number:String);

